src/lib.rs

```diff
@@ -228,12 +228,18 @@ pub struct Header {
     pub face_count: u32,
     pub level_count: u32,
     pub supercompression_scheme: Option<SupercompressionScheme>,
-    dfd_byte_offset: u32,
-    dfd_byte_length: u32,
-    kvd_byte_offset: u32,
-    kvd_byte_length: u32,
-    sgd_byte_offset: u64,
-    sgd_byte_length: u64,
+    pub index: Index,
+}
+
+/// An index giving the byte offsets from the start of the file and byte sizes of the various sections of the KTX file.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct Index {
+    pub dfd_byte_offset: u32,
+    pub dfd_byte_length: u32,
+    pub kvd_byte_offset: u32,
+    pub kvd_byte_length: u32,
+    pub sgd_byte_offset: u64,
+    pub sgd_byte_length: u64,
```

Comment:
What's the downside to just making these public in `Header`?

Reply:
I agree that makes sense for an interface exposed by a system that generates those values for you, but the `ktx2` crate is not such a system; it feels a bit like contorting our interface to benefit the users of one specific downstream consumer. Would it make sense for you to define your own narrower `Header` struct which you internally convert to a `ktx2::Header`? This would also better insulate your interface from breaking changes made in the `ktx2` crate in the future, whereas directly reexporting `ktx2` types makes your interface fragile.

Reply:
No, I agree with that. If you think that these values should just be in the header then we can do that.

Reply:
Alternative idea: reviewing the spec, the index *is* treated as logically independent from the header. Maybe we should split it out into a sibling structure, rather than a member?

Reply:
Yeah, so IMO it both is and is not part of the header. I personally don't think it makes sense to convert it separately but having it as a sub-struct of the header works. I'm happy to go with this PR as-is.